utils/python/CIME/create_test.py

```diff
@@ -95,17 +95,37 @@ def __init__(self, test_names,
         else:
             self._parallel_jobs = parallel_jobs
 
-        if (self._compare or self._generate):
-
+        self._baseline_cmp_name = None
+        self._baseline_gen_name = None
+        self._compare = False
+        self._generate = False
+        if (compare or generate):
+            logging.info("compare is %s generate is %s"%(compare,generate))
             # Figure out what baseline name to use
             if (baseline_name is None):
+                if(compare is not None and compare == str(compare)):
```

Comment:
Yeah, pylint flags type(X) is Y and tells me to use isinstance instead. I kinda disagree for primitive types like str though, the former just reads better to me.